akka-actor/src/main/scala/akka/actor/FaultHandling.scala

```diff
@@ -105,25 +105,52 @@ trait SupervisorStrategyLowPriorityImplicits { this: SupervisorStrategy.type =>
   // the above would clash with seqThrowable2Decider for empty lists
 }
 
+@nowarn("msg=deprecated")
 object SupervisorStrategy extends SupervisorStrategyLowPriorityImplicits {
   sealed trait Directive
 
+  /**
+   * INTERNAL API
+   */
+  @InternalApi
+  private[akka] sealed trait DirectiveWithLevel {
+    def logLevel: LogLevel
+  }
```

Comment:
Directive is sealed so it is safe (bin comp) to add `def logLevel` to that instead of introducing a new trait

Reply:
Escalate does not need a log level as no log is written in that case

Reply:
Fine to just return arbitrary loglevel with a comment saying it is never used, or even throw an exception in that accessor.